mlserver/settings.py

```diff
@@ -38,6 +38,32 @@ class Config:
     """Maximum length (i.e. size) of gRPC payloads."""
     grpc_max_message_length: Optional[int] = None
 
+    # CORS settings
+    enable_cors: bool = False
+    """
+    A list of origins that should be permitted to make 
+    cross-origin requests. E.g. ['https://example.org', 'https://www.example.org']. 
+    You can use ['*'] to allow any origin
+    """
+    allow_origins: Optional[List[str]] = []
+    """
+    A regex string to match against origins that 
+    should be permitted to make cross-origin requests. 
+    e.g. 'https://.*\.example\.org'
+    """
+    allow_origin_regex: Optional[str] = None
+    """Indicate that cookies should be supported for cross-origin requests"""
+    allow_credentials: Optional[bool] = False
+    """A list of HTTP methods that should be allowed for cross-origin requests"""
+    allow_methods: Optional[List[str]] = ["GET"]
+    """A list of HTTP request headers that should be supported for cross-origin requests"""
+    allow_headers: Optional[List[str]] = []
+    """Indicate any response headers that should be made accessible to the browser"""
+    expose_headers: Optional[List[str]] = []
+    """Sets a maximum time in seconds for browsers to cache CORS responses"""
+    max_age: Optional[int] = 600
```

Comment:
nit: would it make sense to group these into a sub-object, e.g. `CORSSettings`? We could replace the `enable_cors` flag for something like this:

```python
class CORSSettings(BaseSettings):
	allow_origins: Optional[List[str]] = []
	...

class Settings(BaseSettings):
	....
	cors_settings: Optional[CORSSettings] = None
```

Since we would have the `cors_settings` as something `None`-able, we could then enable the middleware only when `settings.cors_settings is not None`.

Reply:
I like that idea @adriangonz, I'll make changes to reflect that